.github/workflows/build.yml

```diff
@@ -0,0 +1,53 @@
+name: build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  compile:
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - nightly
+         # - "1.63.0"
+        platform:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/install_deps
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@master
```

Comment:
Don't have much experience with github workflows. How "settled" is `dtolnay/rust-toolchain` i only read about `actions-rs/toolchain` being abandoned and the community is trying to make/find a replacement. Is that it? In any case, even if it changes to a different repository later on we could change that.

Reply:
I am not 100% sure about that either. Here is the relevant issue: https://github.com/actions-rs/toolchain/issues/216. A number of projects switched to it and I also noticed that BurntSushi switched to it for the rust-lang/regex repo ([PR](https://github.com/rust-lang/regex/pull/883)). I believe it is a popular Workflow

Reply:
I think we can just go ahead – we can change it later if its necessary, we don't maneuver our self into a corner either way. I just wanted to know your opinion about it.

Reply:
Okay, sure